### PR TITLE
chore: replace vulnerable rational order stylelint config

### LINKED ISSUE #8450: Security vulnerability in `yargs-parser` via `stylelint-config-carbon`
## What package(s) are you using?

- [x] `stylelint-config-carbon`

## Detailed description

`stylelint-config-carbon`...
* [depends on](https://github.com/carbon-design-system/carbon/blob/67110674cf441d53393a465b2466607ca6e5a30c/config/stylelint-config-carbon/package.json#L30) `stylelint-config-rational-order@^0.1.2`, which
* [depends on](https://github.com/constverum/stylelint-config-rational-order/blob/843a487338a25de246fd2d346fd1caf63f1de8a5/package.json#L33) `stylelint@^9.10.1`, which
* [depends on](https://github.com/stylelint/stylelint/blob/9feb822ca038a6139c8fcca7f535490a86f6eb45/package.json#L62) `meow@^5.0.0`, which 
* [depends on](https://github.com/sindresorhus/meow/blob/e04760557cca1200af91d96a60072f47d6a06aa1/package.json#L49) `yargs-parser@^10.0.0`, which
* [has this security vulnerability](https://www.npmjs.com/advisories/1500).

`yargs-parser` was patched years ago. Its latest version is `yargs-parser@20.2.7`. `meow` upgraded to a patched version of `yargs-parser` a long time ago too; its latest version is `meow@9.0.0`. `stylelint` upgraded to a later version of `meow` a while ago. Its latest version is `stylelint@13.12.0`.

However, `stylelint-config-rational-order`'s latest version is still `stylelint-config-rational-order@0.1.2`. This package's source has not been modified since April 2019 and [has a dozen open pull requests](https://github.com/constverum/stylelint-config-rational-order/pulls). An [issue to have it upgrade its dependency on `stylelint`](https://github.com/constverum/stylelint-config-rational-order/issues/39) has been open since September 2020 and has seen no activity.

I suggest that `stylelint-config-carbon` should stop using `stylelint-config-rational-order` entirely. Another possibility is to fork `stylelint-config-rational-order`, patch it, and use the fork instead. Up to you.

In the meantime, we are going to stop using `stylelint-config-carbon`.

## Steps to reproduce the issue

```shell
npm init
npm install stylelint-config-carbon
npm audit
```

```

                       === npm audit security report ===


                                 Manual Review
             Some vulnerabilities require your attention to resolve

          Visit https://go.npm.me/audit-guide for additional guidance


  Low             Prototype Pollution

  Package         yargs-parser

  Patched in      >=13.1.2 <14.0.0 || >=15.0.1 <16.0.0 || >=18.1.2

  Dependency of   stylelint-config-carbon

  Path            stylelint-config-carbon > stylelint-config-rational-order >
                  stylelint > meow > yargs-parser

  More info       https://npmjs.com/advisories/1500

found 1 low severity vulnerability in 463 scanned packages
  1 vulnerability requires manual review. See the full report for details.
```

### PR DESCRIPTION
**Pull request by emyarod**
Closes #8450

We are replacing `stylelint-config-rational-order` plugin with `stylelint-config-idiomatic-order`, which has fewer dependencies so it is less likely to become a source of future vulnerabilities

### CODE REVIEW

File: packages/components/src/components/button/_button.scss [hunk 1]
```diff
@@ -155,10 +155,10 @@
     &.#{$prefix}--btn--disabled.#{$prefix}--tooltip--a11y::before,
     &.#{$prefix}--btn--disabled.#{$prefix}--tooltip--a11y::after,
     &.#{$prefix}--btn--disabled .#{$prefix}--assistive-text {
-      margin: -1px;
       overflow: hidden;
-      opacity: 0;
```
Comment (tw15egan) [inline packages/components/src/components/button/_button.scss:162]: Just confirming this removal is intentional

Comment (tw15egan) [reply packages/components/src/components/button/_button.scss:162]: Nevermind it's further down, not sure why this one is highlighted different 🙃

**Review by tw15egan — APPROVED** [review]

**Review by tay1orjones — APPROVED** [review]

File: packages/components/src/components/tag/_tag.scss [hunk 2]
```diff
@@ -27,19 +27,19 @@
...
-    // restricts size of contained elements
     max-width: 100%;
     min-height: rem(24px);
-    margin: $carbon--spacing-02;
+    align-items: center;
+    justify-content: center;
     padding: $carbon--spacing-02 $carbon--spacing-03;
-    vertical-align: middle;
-    word-break: break-word;
+    // restricts size of contained elements
```
Comment (jnm2377) [inline packages/components/src/components/tag/_tag.scss]: should this comment have moved too?

Comment (jnm2377) [reply packages/components/src/components/tag/_tag.scss]: I'm unsure if it was a comment for `max-width` or something else. Just kind of confused me. Disregard if it's correct 😅

Comment (emyarod) [reply packages/components/src/components/tag/_tag.scss]: thanks, updated

**Pull request merged**